src/trackHttp.test.js

```diff
@@ -0,0 +1,85 @@
+import {trackPromise, emitter} from './trackHttp';
+import {Emitter} from './tinyemmiter';
+
+
+describe('trackPromise', () => {
+  test('On Initial case, promise fired, promise emitter.emit is called', () => {
+    // Arrange
+    emitter.emit = jest.fn((a,b) => {
+      return;
+    });
+
+    const myPromise =  Promise.resolve().then(() => {
+      return "ok";
+    });
+
+    // Act
+    trackPromise(myPromise);
+
+    // Assert
+    expect(emitter.emit).toHaveBeenCalledTimes(1);
+
+    return myPromise;
+  });
+
+  test('Promise tracked, we got resolve, check that emit is called 2 times', () => {
+    // Arrange
+    emitter.emit = jest.fn((a,b) => {
+      return;
+    });
+
+    const myPromise =  Promise.resolve();
+
+    // Act
+    trackPromise(myPromise);
+
+    // Assert
+    return myPromise.then(() => {
+      expect(emitter.emit).toHaveBeenCalledTimes(2);
+    });
+  });
+
+  test.skip('Promise tracked, we got fail, check that emit is called 2 times', () => {
+    // Arrange
+    expect.assertions(1);
+
+    emitter.emit = jest.fn((a,b) => {
+      return;
+    });
+
+    const myPromise =  Promise.reject();
+
+    // Act
+    trackPromise(myPromise);
+
+    // Assert
+    return myPromise.catch(()=> {
+        expect(emitter.emit).toHaveBeenCalledTimes(2)
+    });
+  });
+
+
+  // Pending promise failed
+  test('Two Promises tracked, we got resolve on both, check that emit is called 4 times', () => {
```

Comment:
In a similar vein to previous cases, I would rename this one to: "when tracking two promises and after both are resolved successfully, emit should have been called four times"